### ISSUE TITLE
[plugins] Changelog check override doesn't always work

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#107485]
See https://github.com/flutter/packages/pull/2312 for an example where even after pushing a non-empty change after changing the PR description, it still failed the check instead of honoring the override. I've seen at least one other case of this failing before as well.

I need to try a test PR in flutter/plugins where I can adjust the script to log more at the same time, to see what it thinks the PR description is as I update the PR.

**Comment by stuartmorgan** [comment]
I added debug logging, and it looks like `CIRRUS_CHANGE_MESSAGE` is now reporting the incremental commit message instead of the PR message, which isn't what it used to do, or what it looks like it's documented to do.

**Comment by stuartmorgan** [comment]
Filed https://github.com/cirruslabs/cirrus-ci-docs/issues/1029

In the short term, a workaround is to repeat the "No CHANGELOG change:" line in incremental commits.

**Comment by stuartmorgan** [comment]
Capturing from discussion in the Cirrus issue: `CIRRUS_CHANGE_MESSAGE` apparently never worked the way I thought I did, and I only thought this worked. (This is likely why I originally thought that re-running would work; I was probably testing against a PR with no incremental commits and it worked, and then later when we saw issues I misremembered what I'd seen before and thus misunderstood the issue).

The new label-based approach proposed there should let us re-run as expected, and work with incremental commits. I'll be validating this more carefully before rolling out a new implementation.
